src/procevent.c

```diff
@@ -778,71 +775,89 @@ static int read_event() {
         recv_flags = 0;
         continue;
       } else if (errno != EINTR) {
-        ERROR("procevent plugin: socket receive error: %d", errno);
+        ERROR("procevent plugin: socket receive error: %s", STRERRNO);
         return -1;
       } else {
         // Interrupt, so just continue and try again
         continue;
       }
     }
+    msg_size = (size_t)status;
+    size_t expected_size = sizeof(struct nlmsghdr) + sizeof(struct cn_msg) +
+                           sizeof(struct proc_event);
+    if (msg_size < expected_size) {
+      ERROR("procevent plugin: received %zu bytes, expected %zu", msg_size,
```

Comment:
IMHO reads better like this (and as next message was already changed):
```suggestion
      ERROR("procevent plugin: received %zu, expected %zu bytes", msg_size,
```